preset/k3s/preset.go

```diff
@@ -82,34 +140,56 @@ func Preset(opts ...Option) gnomock.Preset {
 // P is a Gnomock Preset implementation of lightweight kubernetes (k3s).
 type P struct {
 	Version string `json:"version"`
-	Port    int
 }
 
 // Image returns an image that should be pulled to create this container.
 func (p *P) Image() string {
-	return fmt.Sprintf("docker.io/orlangure/k3s:%s", p.Version)
+	return fmt.Sprintf("docker.io/rancher/k3s:%s", p.Version)
 }
 
 // Ports returns ports that should be used to access this container.
 func (p *P) Ports() gnomock.NamedPorts {
-	port := gnomock.TCP(p.Port)
-	port.HostPort = p.Port
+	port := gnomock.TCP(apiPort)
 
 	return gnomock.NamedPorts{
 		gnomock.DefaultPort: port,
-		KubeconfigPort:      gnomock.TCP(kubeconfigPort),
+		KubeConfigPortName:  gnomock.TCP(kubeconfigPort),
 	}
 }
 
 // Options returns a list of options to configure this container.
 func (p *P) Options() []gnomock.Option {
 	p.setDefaults()
 
+	kubeconfigHttpdJSON, err := json.Marshal(kubeconfigHttpd)
+	if err != nil {
+		log.Fatalf("could not marshal kubeconfig-httpd pod: %s", err)
```

Comment:
I don't think we should use `Fatal` in a library. Since the map doesn't depend on any user input, what about making this JSON inside `init` code, and panicking on error?

Reply:
Ah, I see the pattern we're using now! Thank you, i'll do that instead 👍